mantidproject/mantid
# Change axis limits using scientific notation

**Description of work.**
This pull request makes it possible to change a plot's axis using scientific notation in the figure options menu. This issue was originally reported by James Lord.


**To test:**
1. Plot a figure and open figure options.
2. In the Upper Limit and Lower Limit fields for both axes, check that you can input numbers in both decimal form (e.g. 2.55, -1.2, 3.14159265) and scientific notation (1e-3, 2.5e4).
3. Entering an invalid value such as "e", "-.", "5e." should cause an error dialog to appear.

Fixes #26869 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
